contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
-                two different angles of incidence (a1, a2), then
-                N_measurements = 12 and the order of the values for the various
-                parameter vectors is:
-                
-                * angle_of_incidence: [a1,a1,a1,a1,a1,a1,a2,a2,a2,a2,a2,a2]
-                * pressure: [p1,p1,p1,p2,p2,p2,p1,p1,p1,p2,p2,p2]
-                * temperature: [T1,T2,T3,T1,T2,T3,T1,T2,T3,T1,T2,T3]
-              dimensions:
-                rank: 1
-                dim: [[1, N_measurements]]
-        WINDOW(NXaperture):
-          exists: optional
+            The type of light polarization produced by the source.
+          enumeration: [linear, unpolarized, circular, elliptically, other]
+        device_information(NXfabrication):
           doc: |
-            For environmental measurements, the environment (liquid, vapor
-            etc.) is enclosed in a cell, which has windows both in the
-            direction of the source (entry window) and the detector (exit
-            window) (looking from the sample). In case that the entry and exit
-            windows are not the same type and do not have the same properties,
-            use a second 'WINDOW(MXaperture)' field.
-            
-            The windows also add a phase shift to the light altering the
-            measured signal. This shift has to be corrected based on measuring
-            a known sample (reference sample) or the actual sample of interest
-            in the environmental cell. State if a window correction has been
-            performed in 'window_effects_corrected'. Reference data should be
-            considered as a separate experiment, and a link to the NeXus file
-            should be added in reference_data_link in measured_data.
+            Details about the device information.
+        associated_beam(NX_CHAR):
+          doc: |
+            The path to a beam emitted by this source.
+            Should be named with the same appendix, e.g.,
+            for TYPE=532nmlaser, there should as well be
+            a NXbeam named "beam_532nmlaser" aside of this source
+            instance named "source_532nmlaser"
             
-            The window is considered to be a part of the sample stage but also
-            beam path. Hence, its position within the beam path should be
-            defined by the 'depends_on' field.
-          depends_on:
+            Example: /entry/instrument/beam_532nmlaser
+        device_characteristics(NXdata):
+      detector_TYPE(NXdetector):
+        exists: [min, 1, max, infty]
+        detector_channel_type:
+          exists: recommended
+          enumeration: [single-channel, multichannel]
+        detector_type:
+          exists: recommended
+          doc: |
+            Description of the detector type.
+          enumeration: [CCD, Photomultiplier, Photodiode, Avalanche-Photodiode, Bolometer, Golay Detectors, Pyroelectric Detector]
```

Comment:
Is this an exhaustive list or could there be others?

Reply:
ive added "other" as well as a free-text specfication.